src/Compilers/CSharp/Portable/Compiler/Instrumentation.cs

```diff
@@ -109,6 +115,30 @@ private static bool IsTestMethod(MethodSymbol method)
             // PROTOTYPE (https://github.com/dotnet/roslyn/issues/9811): Make this real. 
             return method.Name.StartsWith("Test");
         }
+
+        internal sealed partial class SynthesizedPrivateImplementationStaticConstructor : SynthesizedGlobalMethodSymbol
+        {
+            internal SynthesizedPrivateImplementationStaticConstructor(SourceModuleSymbol containingModule, PrivateImplementationDetails privateImplementationType, SyntheticBoundNodeFactory factory)
+                : base(containingModule, privateImplementationType, factory.SpecialType(SpecialType.System_Void), WellKnownMemberNames.StaticConstructorName)
+            {
+                this.SetParameters(ImmutableArray<ParameterSymbol>.Empty);
+            }
+
+            public override MethodKind MethodKind => MethodKind.StaticConstructor;
+
+            internal override bool HasSpecialName => true;
+
+            internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+            {
+                SyntheticBoundNodeFactory factory = new SyntheticBoundNodeFactory(this, this.GetNonNullSyntaxNode(), compilationState, diagnostics);
+                factory.CurrentMethod = this;
+
+                BoundStatement mvidInitialization = factory.Assignment(factory.ModuleVersionId(), factory.Property(factory.Property(factory.Typeof(ContainingPrivateImplementationDetailsType), "Module"), "ModuleVersionId"));
```

Comment:
How do we know that "Module" and "ModuleVersionId" properties have the right shape. It feels like we should be using well known members infrastructure to look them up.